fineract-provider/src/main/java/org/apache/fineract/infrastructure/gcm/service/NotificationSenderService.java

```diff
@@ -0,0 +1,128 @@
+import org.apache.fineract.infrastructure.gcm.domain.Message;
+import org.apache.fineract.infrastructure.gcm.domain.Message.Builder;
+import org.apache.fineract.infrastructure.gcm.domain.Message.Priority;
+import org.apache.fineract.infrastructure.gcm.domain.Notification;
+import org.apache.fineract.infrastructure.gcm.domain.Result;
+import org.apache.fineract.infrastructure.gcm.domain.Sender;
+import org.apache.fineract.infrastructure.sms.domain.SmsMessage;
+import org.apache.fineract.infrastructure.sms.domain.SmsMessageRepository;
+import org.apache.fineract.infrastructure.sms.domain.SmsMessageStatusType;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class NotificationSenderService {
+
+	private final DeviceRegistrationRepositoryWrapper deviceRegistrationRepositoryWrapper;
+	private final SmsMessageRepository smsMessageRepository;
+
+	@Autowired
+	public NotificationSenderService(
+			final DeviceRegistrationRepositoryWrapper deviceRegistrationRepositoryWrapper,
+			final SmsMessageRepository smsMessageRepository) {
+		this.deviceRegistrationRepositoryWrapper = deviceRegistrationRepositoryWrapper;
+		this.smsMessageRepository = smsMessageRepository;
+	}
+
+	public void sendNotification(List<SmsMessage> smsMessages) {
+		Map<Long, List<SmsMessage>> notificationByEachClient = getNotificationListByClient(smsMessages);
+		for (Map.Entry<Long, List<SmsMessage>> entry : notificationByEachClient
+				.entrySet()) {
+			this.sendNotifiaction(entry.getKey(), entry.getValue());
+		}
+	}
+
+	public Map<Long, List<SmsMessage>> getNotificationListByClient(
+			List<SmsMessage> smsMessages) {
+		Map<Long, List<SmsMessage>> notificationByEachClient = new HashMap<>();
+		for (SmsMessage smsMessage : smsMessages) {
+			if (smsMessage.getClient() != null) {
+				Long clientId = smsMessage.getClient().getId();
+				if (notificationByEachClient.containsKey(clientId)) {
+					notificationByEachClient.get(clientId).add(smsMessage);
+				} else {
+					List<SmsMessage> msgList = new ArrayList<>(
+							Arrays.asList(smsMessage));
+					notificationByEachClient.put(clientId, msgList);
+				}
+
+			}
+		}
+		return notificationByEachClient;
+	}
+
+	public void sendNotifiaction(Long clientId, List<SmsMessage> smsList) {
+
+		DeviceRegistration deviceRegistration = this.deviceRegistrationRepositoryWrapper
+				.findDeviceRegistrationByClientId(clientId);
+		String registrationId = null;
+		if (deviceRegistration != null) {
+			registrationId = deviceRegistration.getRegistrationId();
+		}
+		for (SmsMessage smsMessage : smsList) {
+			try {
+				Notification notification = new Notification.Builder(
+						GcmConstants.defaultIcon).title(GcmConstants.title)
+						.body(smsMessage.getMessage()).build();
+				Builder b = new Builder();
+				b.notification(notification);
+				b.dryRun(false);
+				b.contentAvailable(true);
+				b.timeToLive(30);
```

Comment:
See if requirement makes sense to make this parameter configurable.

Reply:
No need of configurable parameter for this.